Use role to copy CNI bin

Miouge1 opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
The same tasks are copied over and over, we might as well consolidate to the existing role.

**Special notes for your reviewer**:
`network_plugin: cloud` and `network_plugin: cni` do the same thing, so let's consolidate them into `cni`

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note

```

floryut commented:
Not sure if this will generate error or anything but folder for all network plugins were `owner: kube` instead of `owner: root` in CNI role

Miouge1 commented:
> Not sure if this will generate error or anything but folder for all network plugins were `owner: kube` instead of `owner: root` in CNI role

Each network plugin is a little different, some have `recursive`, different syntax for `mode: ` etc.. I've updated the `cni` role with what looked like the most reasonable & common options. Adding `owner: kube`, but dropping `group:`

floryut commented:
> > Not sure if this will generate error or anything but folder for all network plugins were `owner: kube` instead of `owner: root` in CNI role
> 
> Each network plugin is a little different, some have `recursive`, different syntax for `mode: ` etc.. I've updated the `cni` role with what looked like the most reasonable & common options. Adding `owner: kube`, but dropping `group:`

Don't forget the `Flannel` task ;)

Miouge1 commented:
Added flannel in scope, and also changed the dependency to be in `roles/network_plugin/cni` rather than `roles/kubernetes-apps/network_plugin/cni`

Also removed `roles/kubernetes-apps/network_plugin/cni` since it's exactly the same as what's already in `roles/network_plugin/cni`

LuckySB commented:
/lgtm
